### DISCUSSION TITLE
Best way to use daisyui in a monorepo

### DISCUSSION THREAD

**Discussion by wuarmin** [saadeghi/daisyui#1396] [Q&A] [answered]
Hey,
I'm just trying to figure out the best setup with a good DX for a react monorepo using tailwindcss and the awesome `daisyui`-plugin as the component lib.

My requirement is to have multiple `apps` and a `shared-features` package. Suppose there is a `Login` component in the `shared-features` package. Should the css file be included in the build of the `shared-features` package and imported in the app, or should the css-gen only happen at the apps?

```txt
repo
│
└───apps
│   │
│   └───app1 (has shared-features dependency)
│   |   │   package.json
└───packages
│   │
│   └───shared-features
│   |   │   package.json
│   |   │   src/features/login
│   |   │   ...
```

Thanks and best regards

**Comment by wuarmin** [comment]
I tried [following](https://krzysztofzuraw.com/2022/compiling-tailwind-css-components-in-monorepo/) and it works. 
Do you have better solutions?

**Comment by saadeghi** [comment, marked as answer]
In my opinion, the best way depends on the project.  
If two projects have a lot of common styles, it would be better to generate 1 CSS file and use it for both.  
But if there are a lot of styles that we only need in one of the projects, it would be more efficient to generate 2 CSS files. For example, if one project is a single themed admin panel and the other project is a heavily styled landing page, the admin panel mostly will contain forms and tables but the landing page would have a lot of gradients, animations, etc. with multiple themes. In this case if we generate 2 different CSS files, it will be more efficient for the users.

The link you mentioned is the best way if you want 1 CSS file for 2 projects as far as I know.  

Let me know if you have any questions.
